services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcDatabaseServiceImpl.java

```diff
@@ -75,6 +78,35 @@ public void deleteVpc(String id) throws CacheException {
         this.vpcRepository.deleteItem(id);
     }
 
+    @Override
+    public VpcEntity deleteSubnetIdInVpc(String vpcId, String subnetId) throws ResourceNotFoundException, DatabasePersistenceException, CacheException {
+
+        VpcEntity currentVpcState = null;
+
+        try {
+            this.vpcRepository.startTransaction();
+            currentVpcState = getByVpcId(vpcId);
+            if (currentVpcState == null) {
+                throw new ResourceNotFoundException("Vpc not found : " + vpcId);
+            }
+
+            List<String> subnets = currentVpcState.getSubnets();
+            if (subnets == null || !subnets.contains(subnetId)) {
+                return currentVpcState;
+            }
+
+            subnets.remove(subnetId);
+            currentVpcState.setSubnets(subnets);
+            addVpc(currentVpcState);
+        } catch (ResourceNotFoundException | DatabasePersistenceException | CacheException e) {
+            throw e;
+        } finally {
+            this.vpcRepository.commitTransaction();
+        }
+
+        return currentVpcState;
+    }
+
     @Override
```

Comment:
This is good but there is a convention about the naming. The convention is to call someCache.getTransaction().start(), .commit(), .rollback() instead of wrapping the transaction calls in a method and calling it. It would be better if this code follows the convention.

Another change is required for this to actually work. Ignite configuration declare that this cache must have atomicity mode of TRANSACTIONAL. Here is the declatration.

            <bean class="org.apache.ignite.configuration.CacheConfiguration">
                <!-- Set the cache name. -->
                 <property name="name" value="com.futurewei.alcor.web.entity.vpc.VpcEntity"/>
                <!-- Set the cache mode. -->
                <property name="atomicityMode" value="TRANSACTIONAL"/>
                <!-- Other cache parameters. -->
                <property name="cacheMode" value="PARTITIONED"/>
             </bean>

In kubernetes/services/ignite_config.xml there is this section

 <property name="cacheConfiguration">
under which there is tag <list>. The above declaration should be added under the <list> tag.

Every one needs this update to the ignite config file to not run into  this issue, in addition the iginte database should be deleted. (default is <your_ignite_home>/work). If anyone has valuable data in the ignite DB, let me know we can delete just this cache.

If there are any questions about this extra work, ping me.

Reply:
@pkommoju Thanks. We have updated the transaction naming and added a new cacheConfiguration for VpcRepo.